src/BenchmarkDotNet.Diagnostics.Windows/InliningDiagnoser.cs

```diff
@@ -1,128 +1,56 @@
-        {
-            ProcessIdsUsedInRuns.Clear();
-            statsPerProcess.Clear();
-
-            var sessionName = GetSessionName("JitTracing", benchmark, benchmark.Parameters);
-            session = new TraceEventSession(sessionName);
-            session.EnableProvider(ClrTraceEventParser.ProviderGuid,
-                                   TraceEventLevel.Verbose,
-                                   (ulong)(ClrTraceEventParser.Keywords.JitTracing));
-
-            // The ETW collection thread starts receiving events immediately, but we only
-            // start aggregating them after ProcessStarted is called and we know which process
-            // (or processes) we should be monitoring. Communication between the benchmark thread
-            // and the ETW collection thread is through the statsPerProcess concurrent dictionary
-            // and through the TraceEventSession class, which is thread-safe.
-            Task.Factory.StartNew(() => StartProcessingEvents(benchmark), TaskCreationOptions.LongRunning);
-        }
-
-        public void Stop(Benchmark benchmark, BenchmarkReport report)
-        {
-            // ETW real-time sessions receive events with a slight delay. Typically it
-            // shouldn't be more than a few seconds. This increases the likelihood that
-            // all relevant events are processed by the collection thread by the time we
-            // are done with the benchmark.
-            Thread.Sleep(TimeSpan.FromSeconds(3));
-
-            session.Dispose();
-        }
-
-        public void ProcessStarted(Process process)
-        {
-            ProcessIdsUsedInRuns.Add(process.Id);
-            statsPerProcess.TryAdd(process.Id, null);
-        }
-
-        public void AfterBenchmarkHasRun(Benchmark benchmark, Process process)
-        {
-            // Do nothing
-        }
-
-        public void ProcessStopped(Process process)
-        {
-            // Do nothing
-        }
-
-        public void DisplayResults(ILogger outputLogger)
-        {
-            if (logger.CapturedOutput.Count > 0)
-                outputLogger.WriteLineHeader(new string('-', 20));
-            foreach (var line in logger.CapturedOutput)
-                outputLogger.Write(line.Kind, line.Text);
-        }
-
-        private void StartProcessingEvents(Benchmark benchmark)
+        protected override void AttachToEvents(TraceEventSession session, Benchmark benchmark)
         {
-            var expected = benchmark.Target.Method.DeclaringType.Namespace  + "." +
+            var expected = benchmark.Target.Method.DeclaringType.Namespace + "." +
                            benchmark.Target.Method.DeclaringType.Name;
 
-            logger.WriteLine();
-            logger.WriteLineHeader(new string('-', 20));
-            logger.WriteLineInfo($"{benchmark.DisplayInfo}");
-            logger.WriteLineHeader(new string('-', 20));
+            Logger.WriteLine();
+            Logger.WriteLineHeader(new string('-', 20));
```

Comment:
Could we move `new string('-', 20)` to a static readonly field (e.g. `LogSeparator`)?